arch/risc-v/src/mpfs/mpfs_shead.S

```diff
@@ -0,0 +1,87 @@
+/****************************************************************************
+ * /nuttx-bare/arch/risc-v/src/mpfs/mpfs_shead.S
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <nuttx/irq.h>
+
+#include "chip.h"
+#include "mpfs_memorymap.h"
+#include "riscv_internal.h"
+
+/****************************************************************************
+ * Public Symbols
+ ****************************************************************************/
+
+  /* Imported symbols */
+
+  .extern __trap_vec
+
+  .section .text
+  .global __start_s
+
+/****************************************************************************
+ * Name: __start_s
+ *
+ * Description:
+ *   Supervisor mode start function.
+ *
+ * Input Parameters:
+ *    a0 - hartid
+ *
+ ****************************************************************************/
+
+__start_s:
```

Comment:
Why it is `__start_s` and not `__start`? I think that Makefile still has `$(LD) --entry=__start`

Reply:
This is an error. I have __start defined in my own "miniSBI" which is the M-mode entry point, and this is why I needed a different name here. I need to change this.

Reply:
Fixed